bot/utils/converters.py

```diff
@@ -71,4 +71,39 @@ async def convert(self, ctx: Context, claim: str) -> Claims:
             return Claims.__members__[claim]
         except KeyError:
             raise ConversionError(f'`{claim}` is not a valid Claim')
-        
\ No newline at end of file
+
+class HonorsConverter(Converter):
+    """Sanitize honors argument input for grades_cog"""
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        honors = None
+
+        if argument in ('honors', 'hon'):
+            honors = 'honors'
+        elif argument in ('non-honors', 'non-hon', 'regular', 'normal'):
+            honors = 'non-honors'
+        elif argument == 'all':
+            honors = 'all'
+        else:
+            raise ParserError('Are you sure you used the proper format (ex: math-2060 2017 honors)?')
+
+        return honors
+
+class ProfConverter(Converter):
+    """Sanitize input for prof command in grades_cog"""
+
+    def __init__(self, HonorsConverter):
+        self.honors_sanitizer = HonorsConverter.convert
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        prof = argument
+
+        try:
+            # Test if honors attribute was supplied
+            honors = await self.honors_sanitizer(self, ctx, argument.split()[-1])  # Send last item to the sanitizer
+            prof = " ".join(argument.split()[:-1])  # Remove last word from argument string
+        except ParserError:
+            # Honors attribute was not applied
+            honors = 'all'
+
+        return prof, honors
```

Comment:
We dont want to sanitize input for the prof name, anything that modifies a users input arg can be very confusing. We will use the same behavior we did on grades with the Honors converter